### PR TITLE
Jlc/gating completion first approx

### PR DESCRIPTION
**Pull request by johanv26**
## Description

Add progress and course grade event receiver to send progress_enrollmet data to futurex.

Helpful information to include:

- eox-core and social-auth-app-django added as dependencies.
- signals module created to manage receivers and tasks.
- Tests developed for receivers and tasks.
- The endpoint related to the enrollment progress is in this document on page 14.
https://drive.google.com/file/d/16PXRFCFZuRQMjleQqkm_qSKSfddpwT8g/view
- The data sent is mapped with this form.
image
- Already deployed in stage.

## Testing instructions
Clone the repo. Complete a unit and check logs in sync or async containers. Depends on `CELERY_ALWAYS_EAGER = False`

Configure in settings:
```
    FUTUREX_API_URL = "example"
    FUTUREX_API_CLIENT_ID = "example""
    FUTUREX_API_CLIENT_SECRET = "example"
```
### After
**stage_test**:
Peek 2023-03-24 18-15

## Additional information

Include anything else that will help reviewers and consumers understand the change.
- Jira story: 
https://edunext.atlassian.net/browse/FUTUREX-371

## Checklist for Merge

- [x] Tested in a remote environment
- [ ] Updated documentation
- [x] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

File: eox_nelp/signals/tasks.py [hunk 1]
```diff
@@ -10,76 +10,98 @@
...
     """
     api_client = FuturexApiClient()
-    enrollment_progress_path = "enrollment-progress?_format=json"
-    response = api_client.make_post(enrollment_progress_path, data)
+    response = api_client.enrollment_progress(data)
+
     logger.info(
         "send_futurex_progress --- The data %s  was sent to the url %s. The futurex response was: %s",
         data,
-        f"{api_client.base_url}/{enrollment_progress_path}",
```
Comment (johanv26) [inline eox_nelp/signals/tasks.py]: @andrey-canon why you remove this friendly line?

Comment (andrey-canon) [reply eox_nelp/signals/tasks.py]: beacuse enrollment_progress_path doesn't exist

Comment (johanv26) [reply eox_nelp/signals/tasks.py]: what do u think of keeping in the log the base domain? only api_client.base_url

File: eox_nelp/signals/tests/test_tasks.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+            f"send_futurex_progress --- "
+            f"The data {data} was sent to the futurex service host {service_base_url}. "
+            f"The response was: {service_response}"
+        )
+        futurex_api_client_mock().base_url = service_base_url
+        futurex_api_client_mock().enrollment_progress.return_value = service_response
+
+        with self.assertLogs(tasks.__name__, level="INFO") as logs:
+            _post_futurex_progress(data)
+        futurex_api_client_mock().enrollment_progress.assert_called_with(data)
```
Comment (andrey-canon) [inline eox_nelp/signals/tests/test_tasks.py:116]:
```suggestion
            _post_futurex_progress(data)
    
        futurex_api_client_mock().enrollment_progress.assert_called_with(data)
```

**Pull request merged**